test/tests.js

```diff
@@ -521,6 +521,28 @@ describe("compiler", function () {
     isCallExprStmt(ast.body[5], "module", "export");
   });
 
+  it("should transform default export declaration to expression", function () {
+    import { compile } from "../lib/compiler.js";
+
+    var code = [
+      "export default class {}"
```

Comment:
Hmm, it's specifically the lack of a name which causes the undesired result.  Named declarations are hoisted in a different manner, [in this code](https://github.com/benjamn/reify/blob/master/lib/compiler.js#L371-L379).